src/søknad/steg/2-bosituasjon/Bosituasjon.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { FC, useState } from 'react';
+import { IRoute, Routes } from '../../../routing/Routes';
+
+import { FormattedHTMLMessage, injectIntl, IntlShape } from 'react-intl';
+import Side from '../../../components/side/Side';
+import { useLocation } from 'react-router';
+import { hentNesteRoute } from '../../../routing/utils';
+import { erValgtSvarLiktSomSvar } from '../../../utils/søknad';
+import SeksjonGruppe from '../../../components/gruppe/SeksjonGruppe';
+import LocaleTekst from '../../../language/LocaleTekst';
+import MultiSvarSpørsmål from '../../../components/spørsmål/MultiSvarSpørsmål';
+import { AlertStripeAdvarsel } from 'nav-frontend-alertstriper';
+import { delerSøkerBoligMedAndreVoksne } from './BosituasjonConfig';
+import useSøknadContext from '../../../context/SøknadContext';
+import { IMultiSpørsmål, IMultiSvar } from '../../../models/spørsmal';
+import OmSamboerenDin from './OmSamboerenDin';
+import SøkerSkalFlytteSammenEllerFåSamboer from './SøkerSkalFlytteSammenEllerFåSamboer';
+
+interface Props {
+  intl: IntlShape;
+}
+
+const Bosituasjon: FC<Props> = ({ intl }) => {
+  const { søknad, settSøknad } = useSøknadContext();
+  const { bosituasjon } = søknad;
+  const { søkerDelerBoligMedAndreVoksne } = bosituasjon;
+  const [svarPåHovedspørsmål, settSvarPåHovedspørsmål] = useState('');
+
```

Comment:
**Til Info:** svarPåHovedSpørsmål settes i starten når bruker velger et svar  til spørsmålet "delerSøkerBoligMedAndreVoksne" for første gang. Deretter vil den settes etter at man har resatt samboerDetaljer (hvis kravene inntreffer) for å kunne se om alternativet man velger nå er annerledes fra det man tidligere har valgt. 

**Diskusjon:** 
- Finnes det en bedre måte på å sjekke "forrige" svar på et felt og sammenlikne det med svaret man har nå?

Reply:
Mulig jeg ikke forstår oppsettet, men hvorfor trenger dere en egen useState for dette feltet?

Reply:
Var for å kunne resette bosituasjon i søknad dersom man valgte et nytt alternativ. 
Noen av svarsalternativene vil trigge datovelgere som setter seg i context med en gang, (dagens dato). Har et behov for å holde søknadsobjetket clean, så derfor trenger vi å resette objektet. Bruker variabelen til å holde på "forrige" svar og for å sjekke om søker har valgt et nytt svar, slik at resetting trigges. 

Tar gjerne i mot innspill på navn endringer, eller andre måte rå gjøre det på :) Kan man bruke refs kanskje? Hakke så mye erfaring med det tho :) @henninghaakonsen